cpp/harvest/remid_augment.cc

```diff
@@ -20,63 +20,69 @@
 #include <fstream>
 #include <iostream>
 #include <string>
+#include "FileUtil.h"
 #include "MARC.h"
 
 namespace {
 
 [[noreturn]] void Usage() {
-    std::cerr << "Usage: " << ::progname << " marc_input marc_output\n";
+    std::cerr << "Usage: " << ::progname << "  marc_input marc_output issn_output\n";
     std::cerr << "       marc_input is the marc input file\n";
-    std::cerr << "       marc_output is the marc output file\n";
+    std::cerr << "       marc_output is the marc output file without serial\n";
+    std::cerr << "       issn_output is the text output file for issn that ha\n";
     std::exit(EXIT_FAILURE);
 }
 
+
 } // namespace
 
 int Main(int argc, char **argv) {
-    if (argc < 3)
+    if (argc != 4)
         Usage();
 
+    std::set<std::string> issns;
+    auto input_file(MARC::Reader::Factory(argv[1]));
+    auto marc_output(MARC::Writer::Factory(argv[2]));
+    bool serial;
+    std::ofstream issn_file(argv[3]);
+    while (MARC::Record record = input_file->read()) {
+        serial = false;
+        // It needs a temporary record (new_record) to hold all changes.
+        // If it did a change on current active record, in some cases it will mis-calculate the size of the subfield inside which is
+        // generate a unreadable (error) character.
+        MARC::Record new_record(record);
 
-    const std::string input_filename(argv[1]);
-    const std::string output_filename(argv[2]);
-    const std::string filter_field("084");
-    const char filter_subfield('2');
-    const std::string filter_subfield_value("rvk");
-    const std::string target_field("936");
-    int total_record(0);
-    int total_new_field_added(0);
-
-    auto marc_reader_pointer(MARC::Reader::Factory(input_filename));
-    MARC::Reader * const marc_reader(marc_reader_pointer.get());
-    auto marc_writer(MARC::Writer::Factory(output_filename));
-
-    while (MARC::Record marc_record = marc_reader->read()) {
-        const auto check_field(marc_record.findTag(filter_field));
-        ++total_record;
-        if (check_field == marc_record.end())
-            continue;
-
-        for (const auto &field : marc_record.getTagRange(filter_field)) {
-            if (field.getTag() != filter_field)
-                continue;
+        for (auto &field : record.getTagRange("035")) {
+            const std::string issn(field.getFirstSubfieldWithCode('a'));
+            if (StringUtil::ASCIIToUpper(issn.substr(0, 11)) == "(DE-599)ZDB") {
```

Comment:
Instead of hardcoding the 11 here you could just count the strlen of the prefix that you compare against, to avoid hardcoding